taxonomy-editor-frontend/src/pages/editentry/ListTranslations.jsx

```diff
@@ -1,94 +1,351 @@
+    function handleClose() { setOpen(false); }
+    function handleOpen() { setOpen(true); }
+
+    // Used for addition of a translation language
+    function handleAddTranslation(key) {
+        const newRenderedTranslations = [...renderedTranslations, {'languageCode' : key, 'tags' : []}]
+        setRenderedTranslations(newRenderedTranslations);
+        key = 'tags_' + key; // LC must have a prefix "tags_"
         
-        // Get all tags and its corresponding language code
-        // Tagids need to be recomputed, so shouldn't be rendered
-        // Main language isn't considered, since it's rendered separately
- 
-        if (key.startsWith('tags') && 
-            !key.endsWith(nodeObject.main_language) && 
-            !key.includes('ids')) {
+        // Make changes to the parent NodeObject
+        setNodeObject(prevState => {
+            const newNodeObject = {...prevState};
+            newNodeObject[key] = [];
+            return newNodeObject
+        })
+        setOpen(false);
+    }
+
+    // Used for deleting a translation language
+    function handleDeleteTranslation(key) {
+        const newRenderedTranslations = renderedTranslations.filter(obj => !(key === obj.languageCode))
+        setRenderedTranslations(newRenderedTranslations);
+        key = 'tags_' + key; // LC must have a prefix "tags_"
+        
+        // Make changes to the parent NodeObject
+        setNodeObject(prevState => {
+            const newNodeObject = {...prevState};
+            delete newNodeObject[key];
+            return newNodeObject
+        })
+        setOpen(false);
+    }
+
+    // Changes the translations to be rendered
+    // Dependent on changes occuring in "originalNodeObject"
+    useEffect(() => {
+
+        // Main langauge tags are considered separately, since they have to be rendered first
+        let mainLangTags = []
+        let otherLangTags = []
+
+        Object.keys(originalNodeObject).forEach((key) => {
+        
+            // Get all tags and its corresponding language code
+            // Tagids need to be recomputed, so shouldn't be rendered
+            // Eg: tags_fr
+        
+            if (key.startsWith('tags') && !key.includes('ids')) {
+
+                // If tags are not for the main language, add them to otherLangTags
+                if (!key.endsWith(originalNodeObject.main_language)) {
 
-                // Slice the language code
-                let languageCode = key.slice(-2);
-                renderedTranslations[languageCode] = nodeObject[key]
+                    // Slice the language code
+                    let languageCode = key.slice(-2);
+                    let tobeInsertedObj = {'languageCode' : languageCode, 'tags' : []} // General format for storing tags for different lc's
+                    originalNodeObject[key].map((tag) => (
+                        tobeInsertedObj['tags'].push({
+                            'index' : uuid.v4(), // Give a unique identifier for each tag
```

Comment:
why would we want to create new uuids? even for tags that already exist in the `originalNodeObject`?

Reply:
We create new uuids only when a new node is being fetched. As `originalNodeObject` will never be altered by "non-fetch" code, it has wrapped inside a `useEffect`.

Reply:
well yea, I mean, I was not saying that the `originalNodeObject` would be updated, but that you are mapping (`originalNodeObject[key].map((tag) => {})`) values from the `originalNodeObject` and assigning them a new `index`. I'd assume that those  already have an `index` so is it possible to use the one they already have? On the other hand I'd expect those `index` to come (and be used) from the database, if that's not the case why even bother on defining those in the client, but if that's actually the case why defining them in the client and modifying the one they should already have?